### PR TITLE
Closes #1167 Adds support for stacked line charts

### LINKED ISSUE #1167: Stacked line charts
**Is your feature request related to a problem? Please describe.**

When I'm breaking down a trends graph by a property, that one line splits into many lines. A logical way to represent this would be a stacked line chart

**Describe the solution you'd like**

Add a stacked line chart option https://codepen.io/natenorberg/pen/WwqRar

**Describe alternatives you've considered**

Use the overlapping lines

**Additional context**

***Thank*** you for your feature request - we love each and every one!

### PR DESCRIPTION
**Pull request by Kacppian**
## Changes

- Adds support for stacked and stacked cumulative charts

## Checklist
- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)
- [ ] Cypress E2E tests (if applicable)

### CODE REVIEW

**Comment by EDsCODE** [comment]
Is this ready for review or still WIP? As a tip we usually keep WIP PRs as a draft so the status is super clear!

**Comment by Kacppian** [comment]
It's still in WIP, I'll make it a draft.

**Comment by Kacppian** [comment]
Screenshot 2020-07-13 at 7 19 13 PM

Should I add a `total` to the tooltip? The 21 shown in the screenshot was a bit misleading when I was looking at it.

**Comment by Kacppian** [comment]
Closes #1183

**Comment by Kacppian** [comment]
Closes #1167

**Comment by timgl** [comment]
Did you see any way of adding shading to the surface area below the lines so that it looks more like [this example](https://codepen.io/natenorberg/pen/WwqRar)? Agree it would also be good to have a 'total' in the label

**Comment by Kacppian** [comment]
I'll be able to add that too. I'll make the two changes - 
1. Change the line stacked chart to an area stacked chart
2. Add a total to the tooltips

**Comment by timgl** [comment]
Awesome!

**Review by timgl — COMMENTED** [review]
This looks great! My only comment is that it looks like in the last data point something's going wrong:

image

**Comment by timgl** [comment]
@Kacppian Any ideas why the stack looks weird for the last day? I think it has something to do with those dotted lines

**Comment by Kacppian** [comment]
Sorry, I couldn't find time last week. I'm looking into it and the dotted lines seem to be causing the problem. I'll push my changes shortly.

**Comment by timgl** [comment]
Awesome!

File: frontend/src/lib/colors.ts [hunk 1]
```diff
@@ -97,3 +97,11 @@ export function getChartColors(backgroundColor: string): string[] {
 
     return colorsForBackground(colors[0], colors[1], colors[2])
 }
+
+export function setOpacityToHexColor(hexCode: string, opacity: number): string {
+    // Alpha can be added to hex values to obtain a lighter set of colors
+    // 1A as suffix means the alpha is set to 10% or opacity is 0.1
+    const alpha = Math.floor(255 * opacity).toString(16)
+    const hexWithAlpha = `${hexCode}${alpha}`
+    return hexWithAlpha
```
Comment (mariusandra) [inline frontend/src/lib/colors.ts:106]: Hi, this function fails some cases:

```javascript
// 3-letter codes
setOpacityToHexColor("#fff", 0.4) === "#fff66"

// tiny numbers
setOpacityToHexColor("#ffffff", 0.1)  === "ffffff19"
setOpacityToHexColor("#ffffff", 0.01) === "ffffff2"
```

Comment (Kacppian) [reply frontend/src/lib/colors.ts:106]: Since a lot of things have changed since, would it make sense to raise a new PR for this?

**Comment by Kacppian** [comment]
I'll be submitting another PR, closing it for now.

**Pull request closed without merging**